src/diffusers/pipelines/auto_pipeline.py

```diff
@@ -0,0 +1,908 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team.
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import inspect
+from collections import OrderedDict
+
+from ..configuration_utils import ConfigMixin
+from .controlnet import (
+    StableDiffusionControlNetImg2ImgPipeline,
+    StableDiffusionControlNetInpaintPipeline,
+    StableDiffusionControlNetPipeline,
+)
+from .deepfloyd_if import IFImg2ImgPipeline, IFInpaintingPipeline, IFPipeline
+from .kandinsky import KandinskyImg2ImgPipeline, KandinskyInpaintPipeline, KandinskyPipeline
+from .kandinsky2_2 import KandinskyV22Img2ImgPipeline, KandinskyV22InpaintPipeline, KandinskyV22Pipeline
+from .stable_diffusion import (
+    StableDiffusionImg2ImgPipeline,
+    StableDiffusionInpaintPipeline,
+    StableDiffusionPipeline,
+)
+from .stable_diffusion_xl import (
+    StableDiffusionXLImg2ImgPipeline,
+    StableDiffusionXLInpaintPipeline,
+    StableDiffusionXLPipeline,
+)
+
+
+AUTO_TEXT2IMAGE_PIPELINES_MAPPING = OrderedDict(
+    [
+        ("stable-diffusion", StableDiffusionPipeline),
+        ("stable-diffusion-xl", StableDiffusionXLPipeline),
+        ("if", IFPipeline),
+        ("kandinsky", KandinskyPipeline),
+        ("kdnsinskyv22", KandinskyV22Pipeline),
+        ("controlnet", StableDiffusionControlNetPipeline),
+    ]
+)
+
+AUTO_IMAGE2IMAGE_PIPELINES_MAPPING = OrderedDict(
+    [
+        ("stable-diffusion", StableDiffusionImg2ImgPipeline),
+        ("stable-diffusion-xl", StableDiffusionXLImg2ImgPipeline),
+        ("if", IFImg2ImgPipeline),
+        ("kandinsky", KandinskyImg2ImgPipeline),
+        ("kdnsinskyv22", KandinskyV22Img2ImgPipeline),
+        ("controlnet", StableDiffusionControlNetImg2ImgPipeline),
+    ]
+)
+
+AUTO_INPAINTING_PIPELINES_MAPPING = OrderedDict(
+    [
+        ("stable-diffusion", StableDiffusionInpaintPipeline),
+        ("stable-diffusion-xl", StableDiffusionXLInpaintPipeline),
+        ("if", IFInpaintingPipeline),
+        ("kandinsky-inpaint", KandinskyInpaintPipeline),
```

Comment:
@patrickvonplaten 
for kandinsky, inpainting checkpoint won't work with other pipelines, so I mapped them to a different name, this way our user can still use `AutoPipelineForInpaint.from_pretrained("kandinsky-community/kandinsky-2-2-decoder-inpaint")`, but if they want to load the inpaint checkpoint from `AutoPipelineForText2Image` - they will get an error

I thought about removing the Kandinsky inpainting pipeline from the mapping since it is not linked to any other pipelines. But I think maybe from the user point of view, it is better we make sure that for these 6 diffusion models and these 3 tasks, they will always be able to use AutoPipeline; 

another potential issue with the current method of mapping is that we link the pipelines together - for example, we will allow users to use the sd-xl refiner checkpoint for text-to-image and inpainting pipeline, but it is meant to be used for image-to-image only - should we consider these specific cases and maybe throw a warning?

Reply:
> another potential issue with the current method of mapping is that we link the pipelines together - for example, we will allow users to use the sd-xl refiner checkpoint for text-to-image and inpainting pipeline, but it is meant to be used for image-to-image only - should we consider these specific cases and maybe throw a warning?

Yeah, I think that's the right way to go about it.

Reply:
```suggestion
        ("kandinsky", KandinskyInpaintPipeline),
```

I would actually just leave it like this and then when the user will go from a kandinsky t2i checkpoint to a inpaint one (which won't work in inference) they know why. Think we don't have to add a `-inpaint` here. It's totally ok to throw an error IMO no? Not super happy to have `-inpaint` in a "INPAINTING" mapping. This shows that the API is broken